src/adam_bashforth_adaptive.jl

```diff
@@ -0,0 +1,495 @@
+####################################
+# Explicit Adaptive Adam-Bashforth solvers
+####################################
+# (Hairer & Wanner 1996, Vol I, p.398-400, 421-423)
+# ODE_MS Fixed-step, fixed-order multi-step numerical method
+#   with Adams-Bashforth-Moulton coefficients
+function ode_ms_adaptive(F, y0, tspan, order::Integer;reltol = 1.0e-5,
+                                                      abstol = 1.0e-8,
+                                                      minstep=abs(tspan[1] - tspan[1])/1e18,
+                                                      maxstep=abs(tspan[end] - tspan[1])/2.5,
+                                                      initstep=0.,
+                                                      facmax = 1.5,
+                                                      facmin = .1,
+                                                      fac = .8)
+
+    # initialization of time span, which will have variable length
+    tstart = tspan[1]
+    tfinal = tspan[end]
+    t = typeof(tspan[1])[]
```

Comment:
I ran into some trouble trying to use `eltype`, but I tidied this up by simply setting `t = [tstart]`